### ISSUE TITLE
x/interchainstaking/keeper: AccountBalanceCallback should avoid a runtime panic by invoking sdk.ValidateDenom instead of crashing

### ISSUE THREAD

**Issue by odeke-em** [ingenuity-build/quicksilver#154]
If we examine the code in https://github.com/ingenuity-build/quicksilver/blob/99b1e83ca9c0ede6f54c152a7895870d2ea9c87c/x/interchainstaking/keeper/callbacks.go#L431
we can see that it doesn't return an error but really inside it is comprised of this code in the cosmos-sdk https://github.com/cosmos/cosmos-sdk/blob/a1143138716b64bc4fa0aa53c0f0fa59eb675bb7/types/coin.go#L16-L27 per
![newCoinPanic](https://user-images.githubusercontent.com/4898263/182076152-3f250fa9-a352-4a26-8318-a0d4c37410af.png)

## Remedy
We can invoke sdk.ValidateDenom and then return an error as the signature of AccountBalanceCallback provides for so
```go
if err := sdk.ValidateDenom(denom); err != nil {
    return fmt.Errorf("failed to validate coin denomination %q, got %w", err)
}
coin = sdk.NewCoin(denom, sdk.ZeroInt()) 
```

**Comment by joe-bowman** [comment]
This should be okay, as the AccountBalanceCallback is called on callback from a provable query - so the query must be validated against the remote chain's state for the callback to be triggered (and thus, the denom must be valid at this point) - but that doesn't protect against future unintentional misuse of the callback, so will add this for protection from that. Thank you :)

**Comment by joe-bowman** [comment]
closed by #155
